src/ostorlab/runtimes/lite_local/runtime.py

```diff
@@ -0,0 +1,311 @@
+
+The local runtime requires Docker Swarm to run robust long-running services with a set of configured services.
+"""
+import logging
+import socket
+from typing import List
+from typing import Optional
+
+import click
+import docker
+import tenacity
+from docker.models import services as docker_models_services
+
+from ostorlab import exceptions
+from ostorlab.assets import asset as base_asset
+from ostorlab.cli import console as cli_console
+from ostorlab.cli import docker_requirements_checker
+from ostorlab.cli import install_agent
+from ostorlab.runtimes import definitions
+from ostorlab.runtimes import runtime
+from ostorlab.runtimes.local import agent_runtime
+from ostorlab.runtimes.local.models import models
+
+NETWORK_PREFIX = 'ostorlab_lite_local_network'
+
+logger = logging.getLogger(__name__)
+console = cli_console.Console()
+
+ASSET_INJECTION_AGENT_DEFAULT = 'agent/ostorlab/inject_asset'
+
+class UnhealthyService(exceptions.OstorlabError):
+    """A service by the runtime is considered unhealthy."""
+
+
+class AgentNotInstalled(exceptions.OstorlabError):
+    """Agent image not installed."""
+
+
+class AgentNotHealthy(exceptions.OstorlabError):
+    """Agent not healthy."""
+
+
+def _has_container_image(agent: definitions.AgentSettings):
+    """Check if container image is available"""
+    return agent.container_image is not None
+
+
+def _is_service_type_run(service: docker_models_services.Service) -> bool:
+    """Checks if the service should run once or should be continuously running.
+
+    Args:
+        service: Docker service.
+
+    Returns:
+        Bool indicating if the service is run-once or long-running.
+    """
+    return service.attrs['Spec']['TaskTemplate']['RestartPolicy']['Condition'] == 'none'
+
+
+class LiteLocalRuntime(runtime.Runtime):
+    """Lite Local runtime runs agents locally using Docker Swarm.
+    Lite Local runtime starts all the agents listed in the `AgentRunDefinition`, and then injects the target asset.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        if not docker_requirements_checker.is_docker_installed():
+            console.error('Docker is not installed.')
+            raise click.exceptions.Exit(2)
+        elif not docker_requirements_checker.is_user_permitted():
+            console.error('User does not have permissions to run docker.')
+            raise click.exceptions.Exit(2)
+        elif not docker_requirements_checker.is_docker_working():
+            console.error('Error using docker.')
+            raise click.exceptions.Exit(2)
+        else:
+            if not docker_requirements_checker.is_swarm_initialized():
+                docker_requirements_checker.init_swarm()
+            self._docker_client = docker.from_env()
+
+        self._scan_db: Optional[models.Scan] = None
+
+    @property
+    def name(self) -> str:
+        """Lite Local runtime instance name."""
+        if self._scan_db is not None:
```

Comment:
Actually, this code is not correct. Need to be fixed. Actually the scan id should be passed from the constructor.